cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -453,21 +452,220 @@ private KafkaRebalanceStatus buildRebalanceStatusFromPreviousStatus(KafkaRebalan
+
+        return brokersStats;
+    }
+
+    static class MapAndStatus<T, K> {
+
+        T loadMap;
+
+        K status;
+
+        public T getLoadMap() {
+            return loadMap;
+        }
+
+        public void setLoadMap(T loadMap) {
+            this.loadMap = loadMap;
+        }
+
+        public K getStatus() {
+            return status;
+        }
+
+        public void setStatus(K status) {
+            this.status = status;
+        }
+
+        public MapAndStatus(T loadMap, K status) {
+            this.loadMap = loadMap;
+            this.status = status;
+        }
+    }
+
+    /**
+     * Converts the supplied JSONObject containing the response from the {@link CruiseControlApi#rebalance} or
+     * {@link CruiseControlApi#getUserTaskStatus} methods, into a map linking to a proposal summary map and a broker
+     * load map.
+     *
+     * @param  proposalJson The JSONObject representing the response from the Cruise Control rebalance endpoint.
+     * @return A map containing the proposal summary and broker load maps.
+     */
+    protected static MapAndStatus<ConfigMap, Map<String, Object>> processOptimizationProposal(KafkaRebalance kafkaRebalance, JsonObject proposalJson) {
+
+        JsonArray brokerLoadBeforeOptimization;
+        JsonArray brokerLoadAfterOptimization;
+        if (proposalJson.containsKey(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey()) &&
+                proposalJson.containsKey(CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey())) {
+            brokerLoadBeforeOptimization = proposalJson
+                    .getJsonObject(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey())
+                    .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+            brokerLoadAfterOptimization = proposalJson
+                    .getJsonObject(CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey())
+                    .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+        } else {
+            throw new IllegalArgumentException("The rebalance optimization proposal returned by Cruise Control did not contain broker load information");
+        }
+
+        JsonObject beforeAndAfterBrokerLoad = parseLoadStats(
+                brokerLoadBeforeOptimization, brokerLoadAfterOptimization);
+
+        ConfigMap rebalanceMap = new ConfigMapBuilder()
+                .withApiVersion("v1")
+                .withNewMetadata()
+                .withNamespace(kafkaRebalance.getMetadata().getNamespace())
+                .withName(kafkaRebalance.getMetadata().getName())
+                .withLabels(Collections.singletonMap("app", "strimzi"))
+                .endMetadata()
+                .withData(Collections.singletonMap(BROKER_LOAD_KEY, beforeAndAfterBrokerLoad.encodePrettily()))
+                .build();
+
+        Map<String, Object> optimizationProposal = new HashMap<>();
+        optimizationProposal.put(CruiseControlRebalanceKeys.SUMMARY.getKey(),
```

Comment:
As far as I can see this is a change from the current behavior where the `optimizationResult` given to the `KafkaRebalanceStatusBuilder` is not under an additional summary key it is just the map of the contents of the summary. For example [here](https://github.com/strimzi/strimzi-kafka-operator/blob/1554521a7a0ef5a641717e94ad762834fec89608/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java#L926). So this would change the format of status compare to the current code.

I think you just need to return the summary map as the second argument in the return statement. No need to create a new map for this one entry: `return new MapAndStatus<>(rebalanceMap, proposalJson.getJsonObject(CruiseControlRebalanceKeys.SUMMARY.getKey()).getMap())`